specs/014-kratos-authid/data-model.md

```diff
@@ -0,0 +1,119 @@
+- **State**:
+  - `authId: string` (required once set).
+  - `status: 'synced' | 'missing' | 'duplicate-detected'` (transient, exposed via logs/metrics not persisted yet).
+- **Transitions**:
+  1. **Migrated** – set by TypeORM migration using Kratos admin API.
+  2. **Provisioned** – set during new user creation or via identity resolution endpoint.
+  3. **Exception** – if Kratos lookup fails, domain service emits an exception captured by REST layer for retry.
+
+## Service & DTO Additions
+
+### Identity Resolution REST Module
+
+**Location**: `src/services/api-rest/identity-resolution/`
+
+- `identity-resolution.controller.ts`
+  - `POST /internal/identity/resolve`
+  - Accepts body `IdentityResolutionRequestDto`.
+- `identity-resolution.service.ts`
+  - Orchestrates lookup/provision by invoking `UserService` and `KratosService`.
+  - Emits audit logs and metrics.
+- `identity-resolution.module.ts`
+  - Registers controller/service, applies authentication guard for internal clients.
+
+#### DTOs
+
+**Location**: `src/services/api-rest/identity-resolution/dto/`
+
+```typescript
+export class IdentityResolutionRequestDto {
+  @IsString()
+  @IsNotEmpty()
+  kratosIdentityId!: string;
+}
+
+export class IdentityResolutionResponseDto {
+  @IsUUID()
+  userId!: string;
+  @IsBoolean()
+  created!: boolean; // true when endpoint provisioned a new user
+}
+```
+
+Validation leverages Nest pipe defaults (`class-validator`).
+
+## Infrastructure Layer Updates
+
+### KratosService
+
+**Location**: `src/services/infrastructure/kratos/kratos.service.ts`
+
+- Add method `getIdentityById(id: string): Promise<Identity | undefined>` using `IdentityApi.getIdentity`.
+- Map identity traits to `AgentInfo` via existing `AuthenticationService.createAgentInfo`.
+- Provide resilience (timeouts/retries) consistent with other methods.
+
+## Migration Artifacts
+
+### TypeORM Migration
+
+**Location**: `src/migrations/20251108120000-user-auth-id.ts` (placeholder name)
+
+Responsibilities:
+
+1. **Schema**: `ALTER TABLE user ADD COLUMN authId VARCHAR(191) NULL UNIQUE` (InnoDB index length ≤191 for utf8mb4).
+2. **Backfill**: For each user with verified email:
+   - Query Kratos admin API for matching identity via email.
+   - Persist `authId` when exactly one identity is found.
+   - Record unresolved cases in a dedicated log table or JSON dump (temporary table `user_authid_backfill_audit`).
+3. **Rollback**: Drop unique index/column and audit table if created.
+4. **Idempotency**: Skip updates when `authId` already populated; reuse transaction boundaries.
+
+## Data Validation & Testing
+
+- **Unit Tests**: Add specs under `src/domain/community/user/__tests__/user.service.identity.spec.ts` covering duplicate detection and setter logic.
+- **Integration Tests**: Add `tests/integration/identity-resolution.e2e-spec.ts` verifying REST endpoint behaviours (existing user, auto-provision, error).
```

Comment:
If we have such folders, e.g. `tests/integration`, we need updated set of scripts. Also, there are other integration tests that can be moved there once this is set up.